sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -40,25 +36,18 @@ private Sentry() {
 
     }
 
-    /**
-     * Initialize and statically store a {@link SentryClient} by looking up
-     * a {@link Dsn} and automatically choosing a {@link SentryClientFactory}.
-     *
-     * @return SentryClient
-     */
-    public static SentryClient init() {
-        return init(null, null);
-    }
-
     /**
      * Initialize and statically store a {@link SentryClient} by looking up
      * a {@link Dsn} and using the provided {@link SentryClientFactory}.
      *
      * @param sentryClientFactory SentryClientFactory to use.
      * @return SentryClient
+     *
+     * @deprecated use {@link #init(SentryOptions)}
      */
-    public static SentryClient init(SentryClientFactory sentryClientFactory) {
-        return init(null, sentryClientFactory);
+    @Deprecated
```

Comment:
Is it also needed to deprecate the main init function to get rid of testNG? 
Maybe this could be a separate PR so we can discuss?

Reply:
This method uses the statically initialized lookup instance, which I thought we're trying to shy away from. Therefore I deprecated this method and pointed towards the new way of doing the initialization that supports passing in the configurable sentry options and lookup.

Reply:
We're in a PR that does not deal with TestNG removal (and which I built the TestNG PR on top of), so that's way it seems like mixing 2 things, were the deprecation was just done first.

Reply:
If we at least can mock it all while testing, that'd be enough. I would avoid forcing consumers to simply call an overload with `new Lookup()` as that doesn't add a lot of value to them right now. With regards to introducing breaking changes or `@Deprecated` we can take a look at a bigger picture to where this SDK is going to go in the next few months and decide at that level. For now being able to use those tests using normal mocks as opposed to testing those statics is a huge win.